src/group_theory/subgroup.lean

```diff
@@ -619,6 +619,53 @@ end
 lemma closure_singleton_one : closure ({1} : set G) = ⊥ :=
 by simp [eq_bot_iff_forall, mem_closure_singleton]
 
+@[simp, to_additive] lemma inv_subset_closure (S : set G) : S⁻¹ ⊆ closure S :=
+begin
+  intros s hs,
+  rw [set_like.mem_coe, ←subgroup.inv_mem_iff],
+  exact subset_closure (mem_inv.mp hs),
+end
+
+@[simp, to_additive] lemma closure_inv (S : set G) : closure S⁻¹ = closure S :=
+begin
+  refine le_antisymm ((subgroup.closure_le _).2 _) ((subgroup.closure_le _).2 _),
+  { exact inv_subset_closure S },
+  { simpa only [set.inv_inv] using inv_subset_closure S⁻¹ },
+end
+
+@[to_additive]
+lemma closure_to_submonoid (S : set G) :
+  (closure S).to_submonoid = submonoid.closure (S ∪ S⁻¹) :=
+begin
+  refine le_antisymm _ (submonoid.closure_le.2 _),
+  { intros x hx,
+    refine closure_induction hx (λ x hx, submonoid.closure_mono (subset_union_left S S⁻¹)
+      (submonoid.subset_closure hx)) (submonoid.one_mem _) (λ x y hx hy, submonoid.mul_mem _ hx hy)
+      (λ x hx, _),
+    rwa [←submonoid.mem_closure_inv, set.union_inv, set.inv_inv, set.union_comm] },
+  { simp only [true_and, coe_to_submonoid, union_subset_iff, subset_closure, inv_subset_closure] }
+end
+
+/-- An induction principle for closure membership. If `p` holds for `1` and all elements of
+`k` and their inverse, and is preserved under multiplication, then `p` holds for all elements of
+the closure of `k`. -/
+@[to_additive "An induction principle for additive closure membership. If `p` holds for `0` and all
+elements of `k` and their negation, and is preserved under addition, then `p` holds for all
+elements of the additive closure of `k`."]
+lemma closure_induction'' {p : G → Prop} {x} (h : x ∈ closure k)
+  (Hk : ∀ x ∈ k, p x) (Hk_inv : ∀ x ∈ k, p x⁻¹) (H1 : p 1)
+  (Hmul : ∀ x y, p x → p y → p (x * y)) : p x :=
+begin
+  change x ∈ (closure k).to_submonoid at h,
```

Comment:
I think this one is probably deserving of a lemma right below `coe_to_submonoid`:
```lean
@[simp, to_additive]
lemma mem_to_submonoid (K : subgroup G) (x : G) : x ∈ K.to_submonoid ↔ x ∈ K := iff.rfl
```
Then you can `rw ←mem_to_submonoid` on the line below